src/Avalonia.Controls/WindowIcon.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Avalonia.Platform;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Avalonia.Media.Imaging;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Represents an icon for a window.
+    /// </summary>
+    public class WindowIcon
```

Comment:
I think that `WindowIcon` is more appropriate than `ApplicationIcon` because we can have multiple different icons for each application, but only one per window. Also, instances are used on a `Window` object, not on an `Application` object. Additionally, a true application icon requires configuration outside of Avalonia's domain (i.e. manifest files).